types/swarm/service.go

```diff
@@ -29,6 +30,26 @@ type ServiceMode struct {
 	Global     *GlobalService     `json:",omitempty"`
 }
 
+// UpdateState is the state of a service update.
+type UpdateState string
```

Comment:
I tried having a `FailureAction` type and it turned out to be cumbersome. The client couldn't handle it like a normal string because of the special type. But I can add it back if you think it should have its own type (or remove `UpdateState` - whatever makes more sense).